components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/ApiModelToOAuthConsumerApp.java

```diff
@@ -88,7 +89,7 @@ private void updateIdTokenConfiguration(OAuthConsumerAppDTO consumerAppDTO, IdTo
 
         if (idToken != null) {
             setIfNotNull(idToken.getExpiryInSeconds(), consumerAppDTO::setIdTokenExpiryTime);
-            consumerAppDTO.setAudiences(Optional.ofNullable(idToken.getAudience())
+            consumerAppDTO.setIdTokenAudiences(Optional.ofNullable(idToken.getAudience())
```

Comment:
this should be eother setAudiences or setIdtokenAudience. Send an error response. Check in swagger whether there is a way to deprecate. 

If someone uses setAudiences then have to noti.

Reply:
Check if legacy is turned on

Reply:
Can change logic in the DTO level as well

Reply:
For the moment. Whatever comes to the idtoken can be set for both setAudience and setIdTOkenAudience

Reply:
Reading the config at the API level is not advisable. that is why we are going to use both setters instead of reading the config and then choosing the relevant setter.